bioconda_utils/build.py

```diff
@@ -322,14 +320,45 @@ def build_recipes(
             "SUBDAG=%s (zero-based) but only SUBDAGS=%s "
             "subdags are available")
 
+    failed = []
+    skip_dependent = defaultdict(list)
+
     # Get connected subdags and sort by nodes
     if testonly:
         # use each node as a subdag (they are grouped into equal sizes below)
         subdags = sorted([[n] for n in nx.nodes(dag)])
     else:
-        # take connected components as subdags
-        subdags = sorted(map(sorted, nx.connected_components(dag.to_undirected(
-        ))))
+        # take connected components as subdags, remove cycles
+        subdags = []
+        for cc_nodes in nx.connected_components(dag.to_undirected()):
+            cc = dag.subgraph(sorted(cc_nodes))
+            nodes_in_cycles = set()
+            for cycle in list(nx.simple_cycles(cc)):
```

Comment:
I did some quick tests with the existing packages by clearing the build blacklist to get the cyclic dependency of the Perl packages that caused the troubles before.
Didn't write tests yet, though, sorry :man_facepalming: :face_with_head_bandage: